webapp/components/compat-2021.js

```diff
@@ -34,6 +39,218 @@ class Compat2021 extends PolymerElement {
+        <!-- Chrome/Edge -->
+        <div class="summary-flex-item">
+          <span class="summary-tooltip"></span>
+          <div class="summary-number">--</div>
+          <div class="summary-browser-name"></div>
+        </div>
+        <!-- Firefox -->
+        <div class="summary-flex-item">
+          <span class="summary-tooltip"></span>
+          <div class="summary-number">--</div>
+          <div class="summary-browser-name"></div>
+        </div>
+        <!-- Safari -->
+        <div class="summary-flex-item">
+          <span class="summary-tooltip"></span>
+          <div class="summary-number">--</div>
+          <div class="summary-browser-name"></div>
+        </div>
+      </div>
+`;
+  }
+
+  static get is() {
+    return 'compat-2021-summary';
+  }
+
+  static get properties() {
+    return {
+      stable: {
+        type: Boolean,
+        observer: '_stableChanged',
+      }
+    };
+  }
+
+  _stableChanged() {
+    this.updateSummaryTitles();
+    this.updateSummaryScores();
+  }
+
+  updateSummaryTitles() {
+    let titleDivs = this.$.summaryContainer.querySelectorAll('.summary-browser-name');
+    let titles = this.stable ? STABLE_TITLES : EXPERIMENTAL_TITLES;
+    for (let i = 0; i < titleDivs.length; i++) {
+      titleDivs[i].innerText = titles[i];
+    }
+  }
+
+  async updateSummaryScores() {
+    let scores = await this.calculateSummaryScores(this.stable);
+    let numbers = this.$.summaryContainer.querySelectorAll('.summary-number');
+    let tooltips = this.$.summaryContainer.querySelectorAll('.summary-tooltip');
+    for (let i = 0; i < scores.length; i++) {
+      numbers[i].innerText = scores[i].total;
+      numbers[i].style.color = this.calculateColor(scores[i].total);
+
+      // TODO: Replace tooltips with paper-tooltip.
+      this.updateSummaryTooltip(tooltips[i], scores[i].breakdown);
+    }
+  }
+
+  updateSummaryTooltip(tooltipDiv, scoreBreakdown) {
+    tooltipDiv.innerHTML = '';
+
+    scoreBreakdown.forEach((val, key) => {
+      const keySpan = document.createElement('span');
+      keySpan.innerText = `${key}: `;
+      const valueSpan = document.createElement('span');
+      valueSpan.innerText = val;
+      valueSpan.style.color = this.calculateColor(val * 5);  // Scale to 0-100
+
+      const textDiv = document.createElement('div');
+      textDiv.appendChild(keySpan);
+      textDiv.appendChild(valueSpan);
+
+      tooltipDiv.appendChild(textDiv);
+    });
+  }
+
+  async calculateSummaryScores(stable) {
+    const label = stable ? 'stable' : 'experimental';
+    const url = `${GITHUB_URL_PREFIX}/data/compat2021/summary-${label}.csv`;
+    const csvResp = await fetch(url);
+    if (!csvResp.ok) {
+      throw new Error(`Fetching chart csv data failed: ${csvResp.status}`);
+    }
+    const csvText = await csvResp.text();
+    const csvLines = csvText.split('\n');
+    csvLines.shift();  // We don't need the CSV header.
+    csvLines.pop();  // Trailing empty line.
```

Comment:
```suggestion
    const csvLines = csvText.split('\n').filter(l => l);
    csvLines.shift();  // We don't need the CSV header.
```

That seems less brittle in case the trailing newline is removed.